cocotb/handle.py

```diff
@@ -479,7 +479,16 @@ def value(self):
         return self._value
 
     def __str__(self):
-        return str(self.value)
+        if isinstance(self.value, bytes):
+            # this would be nonsensical for bytes, so skip the deprecation
+            return super().__str__()
+        else:
+
+            warnings.warn(
+                "`str({t})` is deprecated, and in future will return `{t}._path`. "
+                "To get a string representation of the value, use `str({t}.value)`",
+                FutureWarning, stacklevel=2)
+            return str(self.value)
```

Comment:
```suggestion
        warnings.warn(
            "`str({t})` is deprecated, and in future will return `{t}._path`. "
            "To get a string representation of the value, use `str({t}.value)`",
            FutureWarning, stacklevel=2)
            
        if isinstance(self.value, bytes):
            # default to ASCII as is done in StringObject
            return self.value.decode('ascii') 
        else:
            return str(self.value)
```

Per my previous suggestion.